vault/plugin_catalog.go

```diff
@@ -901,7 +929,10 @@ func (c *PluginCatalog) setInternal(ctx context.Context, name string, pluginType
 		return nil, fmt.Errorf("error while validating the command path: %w", err)
 	}
 
-	if symAbs != c.directory {
+	// Format: <plugin_directory>/.oci-cache/<plugin_slug>/<sha256_prefix>
+	shaPrefix := hex.EncodeToString(sha256)[:8]
```

Comment:
This fails in tests, but I'm planning on a quick follow-up to allow registration, so I'll go ahead and merge this and address in a follow-up PR.